jax/api.py

```diff
@@ -1211,8 +1209,8 @@ def linearize(fun: Callable, *primals) -> Tuple[Any, Callable]:
   same linearization point. Moreover if all the input tangent vectors are known
   at once, it can be more efficient to vectorize using `vmap`, as in::
 
-    pushfwd = partial(jvp, f, (x,))
-    y, out_tangents = vmap(pushfwd, out_axes=(None, 0))((in_tangents,))
+    y, f_jvp = api.linearize(f, x)
```

Comment:
Actually, this implementation is much less memory efficient: it requires total memory cost that scales with the depth of the  computation, whereas `vmap`+`jvp` does not. We shouldn't make this change.

Reply:
I got confused by the context, because it talks about applying `f_jvp` multiple times, but the example code was not about `f_jvp`, nor about `linearize`:

```
  This function is mainly useful if you want to apply `f_jvp` multiple times,
  i.e. to evaluate a pushforward for many different input tangent vectors at the
  same linearization point. Moreover if all the input tangent vectors are known
  at once, it can be more efficient to vectorize using `vmap`, as in::

    pushfwd = partial(jvp, f, (x,))
    y, out_tangents = vmap(pushfwd, out_axes=(None, 0))((in_tangents,))
```

I think that the confusing part is the transition at "Moreover", because it talks about "more efficient", but it refers to memory efficiency, while linearize uses more memory but saves computation cost.